CONTRIBUTING.md

```diff
@@ -6,6 +6,12 @@ Loki uses GitHub to manage reviews of pull requests:
 - If you plan to do something more involved, discuss your ideas on the relevant GitHub issue.
 - Make sure to follow the prerequisites below before marking your PR as ready for review.
 
+## Loki Improvement Documents (LIDs)
+
+Before creating a large pull request to change or add functionality, please create a _Loki Improvement Document (LID)_. We use LIDs to discuss and vet ideas submitted by maintainers or the community in an open and transparent way. As of Jan 2023, we are starting with a lightweight LID process and we may add more structure, inspired by Python's [PEP](https://peps.python.org/pep-0001/) and Kafka's [KIP](https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Improvement+Proposals) approaches.
```

Comment:
`large pull request` may not be right term here? (say if we want to refactor or rename something, can also endup huge PR but usually all we need is to have test passes and may not need LID).

May be better to record something like
Any significant changes must go through the _Loki Improvement Document (LID)_

Reply:
The full clause is `Before creating a large pull request to change or add functionality`, which I think covers this point - no?

Reply:
Yea more or less make sense. 
However stressing the `large pull request` seems bit odd to me. Not sure. Main concern is large PR may mean different things even updating set of vendor dependencies can be tagged as `size:XL`. I don't know the right pharse tbh. I feel like we should stress more on "Design decisions, Large feature work, change existing design, etc" rather than size of the pull request.

Not a blocker definitely.

Reply:
I still think the `to change or add functionality` frames this sufficiently. A vendor update doesn't fit into that. I'm happy to change the wording though. I think it's fine right now because folks will only start a LID once it has a sponsor, and a sponsor will reject LIDs like "update to go 1.20" for example because it's not changing or adding functionality.